### ISSUE TITLE
iree_linalg_ext.softmax on tensor<?xf16> causes linker error: undefined symbol: `fmaxf`

### ISSUE THREAD

**Issue by bjacob** [iree-org/iree#14187]
From minimizing https://github.com/openxla/iree/issues/14182.

It seems that any `softmax` on element type `f16` results in a failure to link on x86-64 due to a reference to `fmaxf`.

minimal testcase:

```mlir
func.func @softmax(%in : tensor<?xf16>, %out : tensor<?xf16>) -> tensor<?xf16> {
    %res = iree_linalg_ext.softmax dimension(0) ins(%in : tensor<?xf16>) outs(%out : tensor<?xf16>) -> tensor<?xf16>
    return %res : tensor<?xf16>
}
```

minimal repro:

```
iree-compile /tmp/a.mlir --iree-hal-target-backends=llvm-cpu --iree-llvmcpu-target-triple=x86_64-linux-gnu -o /tmp/a.vmfb --mlir-print-op-on-diagnostic=false
```

resulting error:

```
lld: error: undefined symbol: fmaxf
```

This does not reproduce if `f16` is replaced by `f32`.

On `aarch64` we run into #14186 instead.

**Comment by MaheshRavishankar** [comment]
Should be fixed by https://github.com/openxla/iree/pull/13808

**Comment by allieculp** [comment]
Assigning to @bjacob @hanhanW just to track closure of #13808 and whether this fixes the issue.

**Comment by bjacob** [comment]
Confirmed that this is fixed by #13808. Thanks!
